### PR TITLE
Add custom render functions

### PR DESCRIPTION
**Pull request by havenchyk**
### Description

Add custom render functions to be able to override default markup and keep backward compatibility

### Review

- [x] Verify that all classes affected by the changes have **good** top-level documentation.
- [ ] Verify that new code has been covered with specs (and features if applicable).
- [ ] Verify that a corresponding meaningful changelog entry has been added (`* Add Github pull request template`), for this change if applicable.
- [ ] Test manually.
- [ ] Get two :+1: from code review.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that all new dependencies are included in `package.json`.
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.
- [ ] Verify the branch name starts with `NNNN` where `NNNN` is the issue number.

### CODE REVIEW

File: src/calendar/calendar.jsx [hunk 1]
```diff
@@ -215,57 +247,74 @@ export default class Calendar extends React.Component {
...
+      headerPrevArrow,
+      headerPrevTitle,
+      maxDate,
+      minDate,
+      MonthHeaderComponent = MonthHeader,
+      renderMonthHeader,
+    } = this.props
 
-    return shownNoticeType && <NoticeComponent blockClassName={blockClassName} type={shownNoticeType} />
+    return (
```
Comment (PeterMK85) [inline src/calendar/calendar.jsx]: This can be a render prop as well

File: src/calendar/day.jsx [hunk 2]
```diff
@@ -3,26 +3,37 @@ import formatDate from 'date-fns/format'
...
+    isWorkday,
+    customRender
+  } = props
+  const children = formatDate(date, 'D')
+
+  if (customRender) {
+    return customRender({
+      ...props,
+      children
+    })
```
Comment (PeterMK85) [inline src/calendar/day.jsx]: I would pass the date only without format, because it can be handled on the passed render prop depend on the usecase

**Comment by PeterMK85** [comment]
Personally I would use a clean render prop approach (everywhere pass as children the render prop itself)

As well as I would add TS and migrate to it cos currently there is no easy way to see what props has been passed to the render prop itself.

Anyway for first step it's okay but if no TS planned somehow arguments of render props somehow need to be tracked.

File: src/calendar/day.jsx (hunk 2 above)
Comment (havenchyk) [reply src/calendar/day.jsx]: date is passed here as a part of props, as for children, I wanted to keep backward compatibility, that's the only reason

**Pull request merged**